### ISSUE TITLE
Use IAsyncEnumerable to iterate through nested array in object returned from API

### ISSUE THREAD

**Issue by Sonic198** [dotnet/runtime#101446]
Hello,
I'm using `HttpClient` to integrate with an API which returns data in following format
```json
{
  "eventList": [
    {
      "eventType": "RECEIVED",
      "houseAirWaybill": "AMZ-20240006",
      "iataCode": "SEA",
      "icaoCode": "KSEA",
      "localDatetime": "2024-03-19T10:59:07.908-07:00",
      "localTimezone": "America/Los_Angeles",
      "shipmentType": "CONTAINER",
      "zuluDatetime": "2024-03-19T17:59:07.908Z"
    }
  ]
}
```
It is possible to have hundreds or even thousands of events inside `eventList` so since now we can use `ReadFromJsonAsAsyncEnumerable<TValue>` I was wondering if it would be possible to somehow iterate through that objects in `eventList` array?

**Comment by Tornhoof** [comment]
Are you aware of 
https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.deserializeasyncenumerable
?

**Comment by Sonic198** [comment]
@Tornhoof  but how that would be usable in that case?


> Are you aware of https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.deserializeasyncenumerable ?

**Comment by Tornhoof** [comment]
It depends on the exact structure of your json response. If your array is nested inside another object it won't help.

The reasoning for only root-level IAsyncEnumerable support is here https://github.com/dotnet/runtime/issues/1570#issuecomment-804355005

**Comment by Sonic198** [comment]
> It depends on the exact structure of your json response. If your array is nested inside another object it won't help.
> 
> The reasoning for only root-level IAsyncEnumerable support is here [#1570 (comment)](https://github.com/dotnet/runtime/issues/1570#issuecomment-804355005)

Yeah so as you can see in first post I unfortunately gets one object with nested array of objects in it so I guess there is no easy way of utilizing IAsyncEnumerable to deserialize those data.
